framework/include/materials/FunctorMaterial.h

```diff
@@ -11,11 +11,6 @@
 
 #include "Material.h"
 
-template <typename>
-class FunctorMaterialProperty;
-template <typename>
-class FunctorMaterialPropertyImpl;
-
 /**
  * FunctorMaterials compute functor material properties
  */
```

Comment:
These objects are currently the only ones that can create a `FunctorMaterialProperty`, which is now just an alias for `PiecewiseByBlockLambdaFunctor`. Since they inherit from `Material` they have the nice feature that they are naturally block restrictable which is nice for a user wishing to create a material property that does one thing on one block and one thing on another. Logically, even after all the refactoring in this PR, a `FunctorMaterial` should be to a functor material property what a regular `Material` is to an ordinary old Moose `MaterialProperty`: a place to give "function" definitions of properties

Reply:
With that being said I would say that the answer to your question

> Are functors stored within the subproblem and this object just channels them?

is yes